src/components/DateInput/index.js

```diff
@@ -0,0 +1,172 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled, { css } from 'styled-components'
+import { space, device } from '../../theme'
+import { Input } from '../Input'
+import { Select } from '../Select'
+import { useDeviceInfo } from '../../hooks'
+
+const Container = styled.div`
+  input[type='date'] {
+    ${props =>
+      props.input !== 'native' &&
+      css`
+        display: none;
+      `};
+  }
+
+  .custom {
+    ${props =>
+      props.input !== 'custom' &&
+      css`
+        display: none;
+      `};
+  }
+`
+
+const InputGroup = styled.div`
+  position: relative;
+  display: flex;
+  grid-auto-flow: row;
+
+  @media ${device.mobile} {
+    grid-auto-flow: column;
+  }
+
+  @supports (display: grid) {
+    display: grid;
+    grid-gap: ${space[8]};
+    justify-items: center;
+    margin: auto;
+  }
+`
+
+const InputWrapper = styled.div`
+  width: 100%;
+`
+
+const SelectWrapper = styled.div`
+  width: 100%;
+  padding: 0 ${space[8]};
+
+  @supports (display: grid) {
+    padding: 0;
+  }
+`
+
+function pad(str) {
+  if (str.length === 1) {
+    return (str = '0' + str)
+  }
+  return str
+}
+
+function createDateString(date) {
+  if (!date) return false
+  let d = date
+  if (typeof date.getMonth !== 'function') {
+    d = new Date(date)
+  }
+  let month = '' + (d.getMonth() + 1)
+  let day = '' + d.getDate()
+  const year = d.getFullYear()
+  month = pad(month)
+  day = pad(day)
+  return [year, month, day].join('-')
+}
+
+function createSelectOptions(months) {
+  return months.map((name, i) => ({ value: i.toString(), name }))
+}
+
+function splitDateString(str) {
+  if (!str) return ['', 1, '']
+  return str.split('-')
+}
+
+export function DateInput({ id, label, hideLabel, ...props }) {
+  const { isMobile } = useDeviceInfo()
+  const dateInputRef = React.useRef()
+  const months = createSelectOptions(props.months)
+  const [initialYear, initialMonth, initialDay] = splitDateString(
+    createDateString(props.initialSelectedDate)
+  )
+  const [year, setYear] = React.useState(initialYear || '')
+  const [month, setMonth] = React.useState(initialMonth - 1 || 0)
+  const [day, setDay] = React.useState(initialDay || '')
+
+  React.useEffect(() => {
+    props.onChange(dateInputRef.current.value)
```

Comment:
Should it always fire the onChange, even when the input field is not valid? 🤔

Reply:
Thought about that too, yeah. Think it should always fire, similar to an `onChange` prop on an `<input />` element. I see validation more as a concern of the form that contains this input.